src/Data/Functor/Invariant.hs

```diff
@@ -986,3 +986,17 @@ these, you can derive them using the "Data.Functor.Invariant.TH" module.
 genericInvmap :: (Generic1 f, Invariant (Rep1 f)) => (a -> b) -> (b -> a) -> f a -> f b
 genericInvmap f g = to1 . invmap f g . from1
 #endif
+
+-------------------------------------------------------------------------------
+-- Wrappers
+-------------------------------------------------------------------------------
+
+newtype InvariantProfunctor p a = InvariantProfunctor (p a a)
+
+instance Profunctor p => Invariant (InvariantProfunctor p) where
+  invmap fn1 fn2 (InvariantProfunctor f) = InvariantProfunctor (dimap fn2 fn1 f)
```

Comment:
It would also be worth considering adding an `InvariantProfunctor` version of a function like `invmap2Profunctor`. That is, a top-level function that defined `invmap` that only requires a `Profunctor` constraint. Similarly for `InvariantArrow`.